src/backend/vulkan/src/lib.rs

```diff
@@ -417,16 +414,11 @@ impl hal::Instance<Backend> for Instance {
 
             let str_pointers = cstrings.iter().map(|s| s.as_ptr()).collect::<Vec<_>>();
 
-            let create_info = vk::InstanceCreateInfo {
-                s_type: vk::StructureType::INSTANCE_CREATE_INFO,
-                p_next: ptr::null(),
-                flags: vk::InstanceCreateFlags::empty(),
-                p_application_info: &app_info,
-                enabled_layer_count: layers.len() as _,
-                pp_enabled_layer_names: str_pointers.as_ptr(),
-                enabled_extension_count: extensions.len() as _,
```

Comment:
False alarm. The bug was here: I was combining `extensions` with the set of extensions required by the OpenXR runtime, and didn't spot that the wrong length was being used here. The builder API "fixed" the bug because it takes a slice instead of a length+pointer, avoiding this kind of problem entirely.